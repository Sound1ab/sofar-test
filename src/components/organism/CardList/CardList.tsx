import React from 'react'
import { IEvent } from '../../../interfaces'
import { styled } from '../../../theme'
import { Card } from '../../molecules'

const Style = styled.div`
  position: relative;
`

interface ICardList {
  events: IEvent[]
}

export function CardList({ events }: ICardList) {
  return (
    <Style>
      {events.map(event => (
        <Card
          dayOfMonth={event.date.dayOfMonth}
          dayOfWeek={event.date.dayOfWeek}
          location={event.location}
          month={event.date.month}
        />
      ))}
    </Style>
  )
}
