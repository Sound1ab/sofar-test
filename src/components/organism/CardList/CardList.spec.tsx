import 'jest-dom/extend-expect'
import React from 'react'
import { TOGGLES } from '../../../enums'
import { cleanup, render } from '../../../test-utils'
import { CardList } from './CardList'

const MOCK_EVENTS = [
  {
    date: {
      dayOfWeek: 'Thursday',
      dayOfMonth: 13,
      month: 'Oct',
    },
    location: 'Whitechapel',
    genre: 'rock',
  },
  {
    date: {
      dayOfWeek: 'Thursday',
      dayOfMonth: 13,
      month: 'Oct',
    },
    location: 'Hoxton',
    genre: 'jazz',
  },
]
const MOCK_TOGGLE = TOGGLES.JAZZ

afterEach(cleanup)

describe('CardList.tsx', () => {
  test('CardList matches snapshot', () => {
    const { asFragment } = render(
      <CardList events={MOCK_EVENTS} activeToggle={MOCK_TOGGLE} />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('CardList filters cards based on toggle', async () => {
    const { getAllByTestId } = render(
      <CardList events={MOCK_EVENTS} activeToggle={MOCK_TOGGLE} />
    )

    const cards = getAllByTestId('Card')
    expect(cards.length).toBe(1)
    expect(cards[0]).toHaveTextContent('Hoxton')
    expect(cards[0]).not.toHaveTextContent('Whitechapel')
  })

  test('Cardlist shows all on all toggle', () => {
    const MOCK_ALL_TOGGLE = TOGGLES.ALL
    const { getAllByTestId } = render(
      <CardList events={MOCK_EVENTS} activeToggle={MOCK_ALL_TOGGLE} />
    )

    const cards = getAllByTestId('Card')
    expect(cards.length).toBe(2)
    expect(cards[0]).toHaveTextContent('Whitechapel')
    expect(cards[1]).toHaveTextContent('Hoxton')
  })
})
