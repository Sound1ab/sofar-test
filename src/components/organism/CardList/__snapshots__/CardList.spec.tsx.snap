// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList.tsx CardList matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-ifAKCX LtvRZ"
  >
    <div
      class="sc-htpNat kmHSBC"
      data-testid="Card"
    >
      <div
        class="card-cell card-date"
      >
        <h2
          class="sc-EHOje fBVHip"
          type="h2"
        >
          13
        </h2>
        <h6
          class="sc-bZQynM jheleg"
          type="h6"
        >
          Oct
        </h6>
      </div>
      <div
        class="card-cell card-location"
      >
        <h6
          class="sc-gzVnrw fDlMua"
          type="h6"
        >
          <span
            class="card-highlight"
          >
            Thursday
          </span>
        </h6>
        <h2
          class="sc-htoDjs eWaNkr"
          type="h2"
        >
          Hoxton
        </h2>
      </div>
      <span
        class="card-chevron"
      />
    </div>
  </div>
</DocumentFragment>
`;
